sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ShouldImplementExportedInterfaces.cs

```diff
@@ -81,78 +69,11 @@ protected override void Initialize(SonarAnalysisContext context)
-            if (!(attributeSyntax.Parent?.Parent is ClassDeclarationSyntax attributeTarget))
-            {
-                return null;
-            }
-
-            return semanticModel.GetDeclaredSymbol(attributeTarget);
-        }
-
-        private INamedTypeSymbol GetExportedTypeSyntax(AttributeSyntax attribute, SemanticModel semanticModel)
-        {
-            if (attribute.ArgumentList == null)
-            {
-                return null;
-            }
-
-            var arguments = attribute.ArgumentList.Arguments;
-            if (arguments.Count == 0 ||
-                arguments.Count > 2)
-            {
-                return null;
-            }
-
-            var typeOfExpression = GetTypeFromNamedArgument(arguments) ??
-                GetTypeFromSingleArgumentAttribute(arguments) ??
-                GetTypeFromDoubleArgumentAttribute(arguments, semanticModel);
-
-            var exportedTypeSyntax = (typeOfExpression as TypeOfExpressionSyntax)?.Type;
-            if (exportedTypeSyntax == null)
-            {
-                return null;
-            }
-
-            return semanticModel.GetSymbolInfo(exportedTypeSyntax).Symbol as INamedTypeSymbol;
-        }
-
-        private static ExpressionSyntax GetTypeFromNamedArgument(IEnumerable<AttributeArgumentSyntax> arguments) =>
-            arguments.FirstOrDefault(IsContractTypeNamedArgument)?.Expression;
-
-        private static ExpressionSyntax GetTypeFromDoubleArgumentAttribute(IEnumerable<AttributeArgumentSyntax> arguments,
-            SemanticModel semanticModel)
-        {
-            var firstArgument = arguments.ElementAtOrDefault(0)?.Expression;
-            if (firstArgument != null &&
-                semanticModel.GetConstantValue(firstArgument).Value is string)
-            {
-                // Two arguments, second should be typeof expression
-                return arguments.ElementAtOrDefault(1)?.Expression;
-            }
-
-            return null;
-        }
-
-        private static ExpressionSyntax GetTypeFromSingleArgumentAttribute(IEnumerable<AttributeArgumentSyntax> arguments)
-        {
-            if (arguments.ElementAtOrDefault(1) != null)
-            {
-                return null;
-            }
-
-            // Only one argument, should be typeof expression
-            return arguments.ElementAtOrDefault(0)?.Expression;
-        }
-
-        private static bool IsContractTypeNamedArgument(AttributeArgumentSyntax argument) =>
-            argument.NameColon?.Name.Identifier.ValueText == "contractType";
-
-        private static bool IsOfExportType(ITypeSymbol type, INamedTypeSymbol exportedType) =>
-            type.GetSelfAndBaseTypes()
-                .Union(type.AllInterfaces)
-                .Any(currentType => currentType.Equals(exportedType));
+        protected override string GetIdentifier(AttributeArgumentSyntax argumentSyntax) =>
+            argumentSyntax.NameColon?.Name.Identifier.ValueText;

```

Comment:
Two lines lower, you use `argumentSyntax?`, which implies that this argument might be null, but not here.